### ISSUE TITLE
Ability to pickle the `Catalog`

### ISSUE THREAD

**Issue by Fokko** [apache/iceberg-python#514]
### Feature Request / Improvement

This allows distribution of the Catalog object within Ray.

**Comment by kevinjqliu** [comment]
https://github.com/apache/iceberg-python/issues/513#issuecomment-2009875953

**Comment by kevinjqliu** [comment]
@dev-goyal I see that you've merged [ray-project/ray#46889](https://github.com/ray-project/ray/pull/46889) curious if you think pickling the `Catalog` or `Table` is still valuable to the Ray project

**Comment by kevinjqliu** [comment]
More context:
https://github.com/ray-project/ray/pull/42235#discussion_r1520929199

**Comment by dev-goyal** [comment]
> @dev-goyal I see that you've merged [ray-project/ray#46889](https://github.com/ray-project/ray/pull/46889) curious if you think pickling the `Catalog` or `Table` is still valuable to the Ray project

I think it could help, but for me - I did not find recreating the catalog each time to add much overhead.

**Comment by kevinjqliu** [comment]
> So, it will likely just make the code cleaner is all

Makes sense, the difference is directly pickling the `catalog` object so that it can be distributed to ray workers versus the [current implementation](https://github.com/ray-project/ray/pull/46889/files#diff-b732e60ffc3d7f161d12c866b045be2e354657a122d55aac9d05e9f6c2d2749dR39) which pass the `catalog_kwargs` to reconstruct the catalog at the worker node
